xarray/core/datatree.py

```diff
@@ -1095,7 +1096,8 @@ def from_dict(
         """
 
         # First create the root node
-        root_data = d.pop("/", None)
+        d_cast = cast(dict, d)
+        root_data = d_cast.pop("/", None)
```

Comment:
Mypy is correct here, a Mapping does not include a `.pop` and ignoring the typing errors doesn't solve the bug. 
xarray uses Mappings frequently, for example `xr.core.utils.FrozenDict(dict(a=3)).pop("a", None)` so it's a real issue.

Either you explicitly convert it to dict i.e. `d_cast = dict(d)` or refactor the code to not use the `.pop` since I'm not so sure it's needed.

Reply:
Thanks, I updated it to explicitly covert the type to a dict. The mypy3.9 tests are still passing but the CI Mypy check seems to be returning the same error as before explicitly converting it to a dict.